Buildings/Fluid/HeatExchangers/DXCoils/BaseClasses/CoilDefrostCapacity.mo

```diff
@@ -0,0 +1,257 @@
+
+  parameter Modelica.Units.SI.Power QDefResCap = defCur.QDefResCap
+    "Capacity of resistive defrost element";
+
+  parameter
+    Buildings.Fluid.HeatExchangers.DXCoils.BaseClasses.Types.DefrostTimeMethods
+    defTri=Buildings.Fluid.HeatExchangers.DXCoils.BaseClasses.Types.DefrostTimeMethods.timed
+    "Type of method to trigger the defrost cycle";
+
+  parameter
+    Buildings.Fluid.HeatExchangers.DXCoils.BaseClasses.Types.DefrostOperation
+    defOpe=Buildings.Fluid.HeatExchangers.DXCoils.BaseClasses.Types.DefrostOperation.resistive
+    "Type of defrost method";
+
+  replaceable parameter
+    Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.Defrost
+    defCur constrainedby
+    Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.Defrost
+    "Defrost curve parameter record";
+
+  Modelica.Blocks.Interfaces.RealInput tDefFra(final unit="1", displayUnit="1")
+    "Defrost time period fraction" annotation (Placement(transformation(extent={
+            {-120,90},{-100,110}}), iconTransformation(extent={{-120,100},{-100,
+            120}})));
+
+  Modelica.Blocks.Interfaces.RealInput heaCapMul(
+    final unit="1",
+    displayUnit="1")
+    "Heating capacity multiplier"
+    annotation (Placement(transformation(extent={{-120,60},{-100,80}}),
+      iconTransformation(extent={{-120,70},{-100,90}})));
+
+  Modelica.Blocks.Interfaces.RealInput inpPowMul(
+    final unit="1",
+    displayUnit="1")
+    "Input power multiplier"
+    annotation (Placement(transformation(extent={{-120,30},{-100,50}}),
+      iconTransformation(extent={{-120,40},{-100,60}})));
+
+  Modelica.Blocks.Interfaces.RealInput TConIn(
+    final unit="K",
+    final displayUnit="K",
+    final quantity="ThermodynamicTemperature")
+    "Temperature of air entering indoor condenser unit"
+    annotation (Placement(transformation(extent={{-120,0},{-100,20}}),
+      iconTransformation(extent={{-120,10},{-100,30}})));
+
+  Modelica.Blocks.Interfaces.RealInput XConIn(
+    final unit="kg/kg",
+    displayUnit="kg/kg",
+    final quantity="MassFraction")
+    "Humidity ratio of air entering indoor condenser coil"
+    annotation (Placement(transformation(extent={{-120,-30},{-100,-10}}),
+      iconTransformation(extent={{-120,-30},{-100,-10}})));
+
+  Modelica.Blocks.Interfaces.RealInput pIn(
+    final unit="Pa",
+    displayUnit="Pa",
+    final quantity="Pressure")
+    "Pressure of air entering indoor condenser coil"
+    annotation (Placement(transformation(extent={{-120,-60},{-100,-40}}),
+      iconTransformation(extent={{-120,-60},{-100,-40}})));
+
+  Modelica.Blocks.Interfaces.RealInput TOut(
+    final unit="K",
+    final displayUnit="K",
+    final quantity="ThermodynamicTemperature") "Temperature of outdoor air"
+    annotation (Placement(transformation(extent={{-120,-90},{-100,-70}}),
+      iconTransformation(extent={{-120,-90},{-100,-70}})));
+
+  Modelica.Blocks.Interfaces.RealInput QTot(
```

Comment:
This should be `QTot_flow` for consistency as the top model (and other models) use `QSen_flow` and `QLat_flow`